src/array/unzipWith.ts

```diff
@@ -0,0 +1,50 @@
+import { isNil } from '../predicate';
+
+/**
+ * Unzips a nested array into separate arrays, optionally transforming the unzipped elements.
+ *
+ * This function takes a nested array (an array of arrays) and returns an array where each
+ * element is a tuple containing the corresponding elements from the input arrays. If the input
+ * nested array is null or undefined, or if it is empty, the function returns an empty array.
+ * An optional iteratee function can be provided to transform the unzipped elements.
```

Comment:
```suggestion
 * Unzips an array of arrays, applying an iteratee function to regrouped elements.
```

Why not write this function's description simpler? (I guess this does not lose meanings..)

Reply:
When I was commenting, I referred to the [zip.ts file](https://github.com/toss/es-toolkit/blob/main/src/array/zip.ts). I think I was trying to follow the format too much.